types/encoding.go

```diff
@@ -163,14 +163,17 @@ func (d *decHelper) NextUint64() uint64 {
 	if d.err != nil {
 		return 0
 	}
-	d.Read(d.buf[:])
+	io.ReadFull(d, d.buf[:])
```

Comment:
should we be checking the error returned here?

Reply:
the error is checked internally by the `d.Read` method. I suppose it doesn't hurt though.